clients/filesystem-fuse/src/gravitino_client.rs

```diff
@@ -0,0 +1,277 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+use crate::config::GravitinoConfig;
+use crate::error::{ErrorCode, GvfsError};
+use reqwest::Client;
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::fmt::Debug;
+use urlencoding::encode;
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct Fileset {
+    pub(crate) name: String,
+    #[serde(rename = "type")]
+    pub(crate) fileset_type: String,
+    comment: String,
+    #[serde(rename = "storageLocation")]
+    pub(crate) storage_location: String,
+    properties: HashMap<String, String>,
+}
+
+#[derive(Debug, Deserialize)]
+struct FilesetResponse {
+    code: u32,
+    fileset: Fileset,
+}
+
+#[derive(Debug, Deserialize)]
+struct FileLocationResponse {
+    code: u32,
+    #[serde(rename = "fileLocation")]
+    location: String,
+}
+
+pub(crate) struct GravitinoClient {
+    gravitino_uri: String,
+    metalake: String,
+
+    client: Client,
+}
+
+impl GravitinoClient {
```

Comment:
The client is binded to fileset,   Could you plan to add some abstractions like `loadCatalog` like java client for more extensiable.

Reply:
Currently, I think it's unnecessary to make it so complex. Just providing the required `fileset` API here is sufficient.

Reply:
@jerryshao , WDYT?